internal/security/setup/option/generate.go

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package option
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	config "github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/setup"
+)
+
+// Generate option....
+func Generate() func(*PkiInitOption) (exitCode, error) {
+	return func(pkiInitOption *PkiInitOption) (exitCode, error) {
+
+		if isGenerateNoOp(pkiInitOption) {
+			return normal, nil
+		}
+
+		if statusCode, err := generatePkis(); err != nil {
+			return statusCode, err
+		}
+		generatedDirPath := filepath.Join(getXdgRuntimeDir(), pkiInitGeneratedDir)
+		// Shred the CA private key before deploy
+		caPrivateKeyFile := filepath.Join(generatedDirPath, caServiceName, tlsSecretFileName)
+		if err := secureEraseFile(caPrivateKeyFile); err != nil {
+			return exitWithError, err
+		}
+
+		if err := deploy(generatedDirPath, pkiInitDeployDir); err != nil {
+			return exitWithError, err
+		}
+
+		return normal, nil
+	}
+}
+
+func isGenerateNoOp(pkiInitOption *PkiInitOption) bool {
+	// nop: if the flag is missing or not on
+	return pkiInitOption == nil || !pkiInitOption.GenerateOpt
+}
+
+func generatePkis() (exitCode, error) {
+	baseWorkingDir, err := os.Getwd()
+	if err != nil {
+		return exitWithError, err
+	}
+
+	resourceDirPath := filepath.Join(baseWorkingDir, resourceDirName)
```

Comment:
this should be configurable from the command line, this is hard-coding the utility to require that in the current working directory has `res/pki-setup-vault.json`, etc. in it which is wrong, it should be configurable where to find these things, either through a configuration.toml file or through a command line option

Reply:
will be through a configuration .toml file...in this phase, we only focus on the functionality. will address this issue in another PR.

Reply:
#1767  is filed for this issue